lib/packet.ml

```diff
@@ -125,22 +130,41 @@ type extension_type =
   | CLIENT_AUTHZ                           [@id 7]  (*RFC5878*)
   | SERVER_AUTHZ                           [@id 8]  (*RFC5878*)
   | CERT_TYPE                              [@id 9]  (*RFC6091*)
-  | ELLIPTIC_CURVES                        [@id 10] (*RFC4492*)
+  | SUPPORTED_GROUPS                       [@id 10] (*RFC4492, RFC8446*)
   | EC_POINT_FORMATS                       [@id 11] (*RFC4492*)
   | SRP                                    [@id 12] (*RFC5054*)
   | SIGNATURE_ALGORITHMS                   [@id 13] (*RFC5246*)
-  | USE_SRP                                [@id 14] (*RFC5764*)
+  | USE_SRTP                               [@id 14] (*RFC5764*)
   | HEARTBEAT                              [@id 15] (*RFC6520*)
   | APPLICATION_LAYER_PROTOCOL_NEGOTIATION [@id 16] (*RFC7301*)
   | STATUS_REQUEST_V2                      [@id 17] (*RFC6961*)
   | SIGNED_CERTIFICATE_TIMESTAMP           [@id 18] (*RFC6962*)
   | CLIENT_CERTIFICATE_TYPE                [@id 19] (*RFC7250*)
   | SERVER_CERTIFICATE_TYPE                [@id 20] (*RFC7250*)
-  | PADDING                                [@id 21] (*draft-ietf-tls-padding*)
+  | PADDING                                [@id 21] (*RFC7685*)
   | ENCRYPT_THEN_MAC                       [@id 22] (*RFC7366*)
-  | EXTENDED_MASTER_SECRET                 [@id 23] (*draft-ietf-tls-session-hash*)
-  | SESSIONTICKET_TLS                      [@id 35] (*RFC4507*)
+  | EXTENDED_MASTER_SECRET                 [@id 23] (*RFC7627*)
+  | TOKEN_BINDING                          [@id 24] (*RFC8472*)
+  | CACHED_INFO                            [@id 25] (*RFC7924*)
+  | TLS_LTS                                [@id 26] (*draft-gutmann-tls-lts*)
+  | COMPRESSED_CERTIFICATE                 [@id 27] (*draft-ietf-tls-certificate-compression*)
+  | RECORD_SIZE_LIMIT                      [@id 28] (*RFC8449*)
+  | PWD_PROTECT                            [@id 29] (*RFC-harkins-tls-dragonfly-03*)
+  | PWD_CLEAR                              [@id 30] (*RFC-harkins-tls-dragonfly-03*)
+  | PASSWORD_SALT                          [@id 31] (*RFC-harkins-tls-dragonfly-03*)
+  | SESSION_TICKET                         [@id 35] (*RFC4507*)
```

Comment:
not sure what you suggest here -- do you really think it is worth to point to specific sections for all constructors? I find this a bit excessive (and additionally, in most cases the constructors have been defined in much earlier RFCs). The comments are mostly from https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml (which, in case you want to lookup/debug existing magic values, you should read / be aware of anyways).